typescript/infra/src/config/agent.ts

```diff
@@ -231,32 +231,36 @@ export type RustConnection =
       url: string;
     }
   | { type: ConnectionType.Ws; url: string }
-  | { type: ConnectionType.HttpQuorum; urls: string };
+  | { type: ConnectionType.HttpQuorum; urls: string }
+  | { type: ConnectionType.HttpFallback; urls: string };
 
 export type RustCoreAddresses = {
   mailbox: types.Address;
   interchainGasPaymaster: types.Address;
   validatorAnnounce: types.Address;
 };
 
-export type RustChainSetup = {
+export interface RustChainSetupBase {
   name: ChainName;
   domain: number;
-  signer?: RustSigner | null;
+  signer?: RustSigner;
   finalityBlocks: number;
   addresses: RustCoreAddresses;
   protocol: 'ethereum' | 'fuel';
-  connection: RustConnection;
+  connection?: RustConnection;
   index?: { from: number };
-};
+}
+
+export interface RustChainSetup extends RustChainSetupBase {
+  signer: RustSigner;
+  connection: RustConnection;
+}
 
 export type RustConfig = {
-  chains: Partial<ChainMap<RustChainSetup>>;
-  // TODO: Separate DBs for each chain (fold into RustChainSetup)
-  db: string;
-  tracing: {
-    level: string;
-    fmt: 'json';
+  chains: Partial<ChainMap<RustChainSetupBase>>;
+  tracing?: {
```

Comment:
It has been removed. It makes no sense to put an objectively wrong default here.

Reply:
Do we default to anything for the db path?

Reply:
No, it will just tell them they must set it. Which I think is reasonable. The alternative would be to just put it in a local directory but they already have to set it right now so it does not really change any functionality. If you want a default set let's set it in a new PR.